labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,7 +60,15 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case s: String => s"A string with length ${s.length()}"
+      case i: Int if i > 0 => "A positive integer"
+      case p: Person => s"A person with name: ${p.name}"
+      case seq: Seq[_] if seq.length > 10 => "Seq with more than 10 elements"
+      case seq: Seq[_] if seq.length >= 3 => s"first: ${seq(0)}, second: ${seq(1)}, rest: ${seq.drop(2)}"
+      case null => "A null value"
+      case _ => "Some Scala class"
+    }
```

Comment:
I'm actually going to push back on this a bit.

> Why do we give even simple variables descriptive names?

We do this because while the meaning of a simple variable may be clear during its declaration, its scope may be large - and it may end up being used far away from that declaration

If we believe this reasoning, then given the short scope of these variables (declaration right next to use), and the fact that the static type system is forcing us to identify their type explicitly,
I think using `s`, `i`, `p` isn't really a problem here.